cpp/src/arrow/acero/asof_join_node_test.cc

```diff
@@ -16,6 +16,7 @@
 // under the License.
 
 #include <gmock/gmock-matchers.h>
+#include <iostream>  // nocommit
```

Comment:
Is this required? Also, what is "nocommit" for?

Reply:
whoops - this was for some cout debugging, will remove